weblate/utils/search.py

```diff
@@ -520,6 +521,30 @@ def component_field(self, text: str, context: dict) -> Q:
             translation__component__name__icontains=text
         )
 
+    def labels_count_field(self, text: str, context: dict) -> Q:
+        from django.db.models import Count
+
+        from weblate.trans.models import Unit
+
+        try:
+            value = self.convert_labels_count(text)
+        except ValueError as error:
+            raise ValueError(f"Invalid value for labels_count: {text}") from error
+
+        suffix = OPERATOR_MAP.get(self.operator)
+        if suffix == "substring":
+            suffix = "exact"
+        filter_kwargs = {f"labels_count__{suffix}": value}
+
+        return Q(
+            id__in=Unit.objects.annotate(
+                labels_count=Count("source_unit__labels", distinct=True)
+                + Count("labels", distinct=True)
+            )
```

Comment:
This will have horrible performance wise on larger sites as it filters *all* units with more labels to then search within a selected scope.